ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseUri.java

```diff
@@ -100,9 +102,18 @@ protected PubsubMessage processElement(PubsubMessage message)
           zip(GENERIC_URI_SUFFIX_ELEMENTS, uri.substring(GENERIC_URI_PREFIX.length()).split("/")));
     } else if (uri.startsWith(StubUri.PREFIX)) {
       payload = StubUri.parse(uri, attributes);
+
+      if (attributes.get("messageId") != null) {
+        // convert PubSub message ID to document ID
+        UUID documentId = UUID
+            .nameUUIDFromBytes(attributes.get("messageId").getBytes(StandardCharsets.UTF_8));
+        attributes.put(Attribute.DOCUMENT_ID, documentId.toString().toLowerCase());
+      }
     } else {
       throw new InvalidUriException("Unknown URI prefix");
     }
+
+    attributes.remove("messageId");
```

Comment:
I guess we'll probably want to remove messageId from the attributes?

Reply:
That's probably a good idea, though worth adding a comment here about the fact that we expect messageId to be relevant only for coverting to docId, and thus isn't useful past this point.